conans/client/subsystems.py

```diff
@@ -128,6 +128,7 @@ def deduce_subsystem(conanfile, scope):
     - Aggregation of envfiles: to map each aggregated path to the subsystem
     - unix_path: util for recipes
     """
+    scope = scope or "build"  # let's assume build context is scope=None
```

Comment:
Might break if some user was using ``scope=""``, which was not crashing before. Better check ``None``